src/Source/AbstractPropertySourceIterator.php

```diff
@@ -0,0 +1,177 @@
+     */
+    protected $fields = [];
+
+    /**
+     * @param array<string> $fields Fields to export
+     */
+    public function __construct(array $fields, string $dateTimeFormat = 'r')
+    {
+        $this->propertyAccessor = PropertyAccess::createPropertyAccessor();
+        $this->fields = $fields;
+
+        $this->dateTimeFormat = $dateTimeFormat;
+    }
+
+    public function current()
+    {
+        $current = $this->iterator->current();
+
+        return $this->getCurrentData($current);
+    }
+
+    public function next(): void
+    {
+        $this->iterator->next();
+    }
+
+    public function key()
+    {
+        return $this->iterator->key();
+    }
+
+    public function valid(): bool
+    {
+        return $this->iterator->valid();
+    }
+
+    abstract public function rewind(): void;
+
+    public function setDateTimeFormat(string $dateTimeFormat): void
+    {
+        $this->dateTimeFormat = $dateTimeFormat;
+    }
+
+    public function getDateTimeFormat(): string
+    {
+        return $this->dateTimeFormat;
+    }
+
+    /**
+     * NEXT_MAJOR: Change the method visibility to private.
+     *
+     * @return string An ISO8601 duration
+     */
+    public function getDuration(\DateInterval $interval): string
+    {
+        $datePart = '';
+        foreach (self::DATE_PARTS as $datePartAttribute => $datePartAttributeString) {
+            if ($interval->$datePartAttribute !== 0) {
+                $datePart .= $interval->$datePartAttribute.$datePartAttributeString;
+            }
+        }
+
+        $timePart = '';
+        foreach (self::TIME_PARTS as $timePartAttribute => $timePartAttributeString) {
+            if ($interval->$timePartAttribute !== 0) {
+                $timePart .= $interval->$timePartAttribute.$timePartAttributeString;
+            }
+        }
+
+        if ('' === $datePart && '' === $timePart) {
+            return 'P0Y';
+        }
+
+        return 'P'.$datePart.('' !== $timePart ? 'T'.$timePart : '');
+    }
+
+    /**
+     * @param object|array $current
+     */
+    protected function getCurrentData($current): array
+    {
+        $data = [];
+        foreach ($this->fields as $key => $field) {
+            if (\is_string($field)) {
+                $name = \is_string($key) ? $key : $field;
+                $propertyPath = $field;
+            } else {
+                throw new \TypeError('Unsupported field type. Field should be a string.');
+            }
+
+            try {
+                $propertyValue = $this->propertyAccessor->getValue($current, new PropertyPath($propertyPath));
+
+                $data[$name] = $this->getValue($propertyValue);
+            } catch (UnexpectedTypeException $e) {
+                // Non existent object in path will be ignored but a wrong path will still throw exceptions
+                $data[$name] = null;
+            }
+        }
+
+        return $data;
+    }
+
+    /**
+     * @param mixed $value
+     *
+     * @return string|null
```

Comment:
Why is this not a return type hint?

Reply:
To be BC. The method was already existing.